### PR TITLE
#4605 SonarCloud fixes

### LINKED ISSUE #4605: Fix SonarCloud issues
In the `keptn/keptn` repo, we have a huge amount of SonarCloud issues accumulated over the past time.
There are currently:

- 16 Bugs
- 2 Vulnerabilities
- 6 Security Hotspots
- 437 Code Smells which results in 5d of technical debt (calculated by SonarCloud)
- 8.8% code duplications

With little work, those numbers can and should be reduced drastically.

### PR DESCRIPTION
**Pull request by mowies**
**This PR**
* fixes many of the current sonarcloud code smells

Closes #4605

Signed-off-by: Moritz Wiesinger <moritz.wiesinger@dynatrace.com>

### CODE REVIEW

**Comment by warber** [comment]
changes in go code looks good to me.
waiting for the final approval from @laneli regarding the bridge code...

pls resolve the conflicts

**Review by laneli — DISMISSED** [review]
Approved from my side.

**Review by laneli — APPROVED** [review]
Merge also approved

**Pull request merged**
